provider/docker_test.go

```diff
@@ -2271,3 +2275,93 @@ func TestSwarmTaskParsing(t *testing.T) {
 		}
 	}
 }
+
+type fakeTasksClient struct {
+	dockerClient.APIClient
+	tasks []swarm.Task
+	err   error
```

Comment:
In fact, as it is used in return statement of `TaskList`, we may leave it as is.